subset_vds.py

```diff
@@ -0,0 +1,340 @@
+
+
+def subset_by_samples(
+    input_vds: VariantDataset, subset_sample_list: list[str]
+) -> VariantDataset:
+    """subset a vds by samples
+
+    Args:
+        input_vds (VariantDataset): the input vds to filter on
+        subset_sample_list (list[str]): the list of sampels to retain in the vds
+
+    Returns:
+        VariantDataset: the subset vds
+    """
+    subset_vds: VariantDataset
+    subset_vds = filter_samples(input_vds, subset_sample_list)
+    return subset_vds
+
+
+def subset_by_locus(
+    parsed_locus: list[IntervalExpression], input_vds: VariantDataset
+) -> VariantDataset:
+    """subset a vds by locus
+
+    Args:
+        parsed_locus (list[IntervalExpression]): a list of locuses to filter the vds down to
+        input_vds (VariantDataset): the input vds to filter
+
+    Raises:
+        ValueError: throws an error if all rows are removed by filtering
+
+    Returns:
+        VariantDataset: the subset vds
+    """
+    subset_vds: VariantDataset
+    subset_vds = filter_intervals(input_vds, parsed_locus)
+    if subset_vds.variant_data.count_rows() == 0:
+        raise ValueError(
+            f"No rows remain after applying the following locus filters: {parsed_locus}"
+        )
+    return subset_vds
+
+
+def write_outputs(
+    output_formats: list[str],
+    subset_vds: VariantDataset | None,
+    subset_sample_list: list[str] | None,
+    infile_name: str,
+) -> None:
+    """Writes the vds out in the specified formats
+
+    Args:
+        output_formats (list[str]): list of output formats to write
+        subset_vds (VariantDataset | None): the resulting vds after subsetting by samples and / or intervals
+        subset_sample_list (list[str] | None): the samples left in the output data, when subsetting by samples
+        infile_name (str): the name of the input vds, to append '_subset' to
+        is_test (bool): controls whether to write to test or not
+
+    Raises:
+        FileExistsError: throws an error if any of the proposed output paths exist, as it will not overwrite them
+    """
+    if "vds" in output_formats:
+        subset_vds.write(output_path(f"{infile_name}_subset"))
+
+    subset_dense_mt: MatrixTable | Table | Any = to_dense_mt(subset_vds)
+    subset_dense_mt = split_multi_hts(subset_dense_mt)
+
+    for format in output_formats:
+        if format == "bed":
+            export_plink(
+                subset_dense_mt,
+                output_path(f"{infile_name}_subset"),
+                call=subset_dense_mt.LGT,
+                ind_id=subset_dense_mt.s,
+            )
+        if format == "vcf":
+            if "gvcf_info" in subset_dense_mt.entry:
+                subset_dense_mt = subset_dense_mt.drop("gvcf_info")
+            export_vcf(
+                subset_dense_mt,
+                output_path(f"{infile_name}_subset.vcf.bgz"),
+                tabix=True,
+            )
+
+    if subset_sample_list:
+        if to_path(output_path("subset_samples_file.txt")).exists():
```

Comment:
This is a bit of a nit, but you might want to move this existence check earlier in the process, else you could potentially be in a position where the subset itself has been written, but you raise an exception before writing the sample list.

That's probably not going to happen, and you can pull the Sample subset easily from the VDS itself, so not a real issue

Reply:
Fair point though, changed that